### PR TITLE
Calculate dictionaries by session.

### PR DESCRIPTION
**Pull request by artem-davidovich**

### CODE REVIEW

File: src/main/kotlin/com/exactpro/th2/actuibackend/schema/SchemaParser.kt [hunk 1]
```diff
@@ -245,10 +246,62 @@ class SchemaParser(private val context: Context) {
...
+            }
+            .flatMap { it }
+            .mapNotNull { resource ->
+                resource.second?.get("metadata")?.elements()?.asSequence()?.map { resource.first to it }
+            }
+            .flatMap { it }
+            .filter { it.second?.get("field-name")?.textValue() == "session_alias" }
+            .filter { it.second?.get("expected-value")?.textValue() == session }
+            .mapNotNull { it.first }
+            .toList()
```
Comment (artem-davidovich) [inline src/main/kotlin/com/exactpro/th2/actuibackend/schema/SchemaParser.kt]: @Paleontolog Can we simplify this pipeline?

Comment (artem-davidovich) [reply src/main/kotlin/com/exactpro/th2/actuibackend/schema/SchemaParser.kt]:
```
{
  "kind": "Th2Box",
  "name": "boxName",
  "spec": {
    "pins": [
      {
        "name": "pinName",
        "filters": [
          {
            "metadata": [
              {
                "field-name": "session_alias",
                "expected-value": "value",
              }
            ]
          }
        ]
      }
    ]
  }
}
```

Comment (Paleontolog) [reply src/main/kotlin/com/exactpro/th2/actuibackend/schema/SchemaParser.kt]: We can use findValue and findValues that recursively finds all fields with specified name and get in values.
As example:

fun checkValuesContains(jsonNode: JsonNode, field: String, value: String): Boolean {
      return jsonNode.findValues(field).any { it.textValue() == value }
}
 val boxesBySession = jsonTree.get("resources").elements().asSequence()
            .filter {
                checkValuesContains(it, "field-name", "session_alias")
                        && checkValuesContains(it, "expected-value", session)
            }.mapNotNull {
                it.get("name")?.textValue()
            }.toList()

File: src/main/kotlin/com/exactpro/th2/actuibackend/schema/SchemaParser.kt [hunk 2]
```diff
@@ -245,10 +246,62 @@ class SchemaParser(private val context: Context) {
...
+            .toSet()
+
+        if (linkedBoxes.isEmpty()) {
+            return getDictionaries()
+        }
+
+        return jsonTree.findParents("kind").asSequence().filter {
+            it?.get("kind")?.textValue() == "Th2Link"
+        }
+            .mapNotNull { it?.get("spec")?.get("dictionaries-relation") }
```
Comment (Paleontolog) [inline src/main/kotlin/com/exactpro/th2/actuibackend/schema/SchemaParser.kt]: we can use findValue instead of chain of get

**Comment by artem-davidovich** [comment]
Requires testing

**Review by Paleontolog — APPROVED** [review]

**Comment by artem-davidovich** [comment]
Requires testing

**Review by Paleontolog — APPROVED** [review]

**Pull request merged**